flink-runtime/src/main/java/org/apache/flink/runtime/operators/shipping/OutputEmitter.java

```diff
@@ -143,14 +148,25 @@ public OutputEmitter(ShipStrategyType strategy, TypeComparator<T> comparator, Pa
 	
 	// --------------------------------------------------------------------------------------------
 
+	private int[] forward() {
+		if (this.channels == null) {
```

Comment:
This method preserves the behavior of `FORWARD` only if `numberOfChannels` is always 0 when `FORWARD` is configured which is out of control of the `OutputEmitter`.
Should we add a check somewhere (e.g., in this condition to reduce the overhead) to ensure that `numberOfChannels` is acutally 0 for `FORWARD`?